lint/base_linter/node_linter.py

```diff
@@ -47,6 +47,22 @@ def smart_paths_upwards(start_dir):
     )
 
 
+def is_yarn_project(path, manifest):
+    # type: (str, Dict[str, Any]) -> bool
+    package_manager = manifest.get('packageManager')
+    if isinstance(package_manager, str):
+        if package_manager.startswith('@'):
+            package_manager = package_manager[1:]
+        name = package_manager.split('@', 1)[0]
+        return name == 'yarn'
+
+    yarn_files = ['yarn.lock', '.yarnrc.yml', '.yarnrc']
+    if [file for file in yarn_files if os.path.exists(os.path.join(path, file))]:
```

Comment:
Various ways to make this efficiently stop on the first match. Typically `if any(os.path.exists(...) for file in yarnfiles)`.

Reply:
Now uses `any` with a generator expression.